project-addons/block_invoices/models/sale_order.py

```diff
@@ -66,4 +66,11 @@ def _action_confirm(self):
             message = _('Order blocked. Approve pending')
             raise exceptions.Warning(message)
 
+        if not self.pricelist_id.name.startswith('PVI'):
+            for line in self.order_line:
+                if line.margin_perc_rappel < 10 and not line.deposit and not line.promotion_line and line.product_id.state != 'end' and line.product_id.categ_id.id not in (392, 393, 468) and not self.allow_confirm_blocked_magreb:
```

Comment:
A modo de mejora futura o cuando os planteeis este tipo de comprobaciones podéis echar un ojo a https://github.com/OCA/server-ux/tree/11.0/base_tier_validation y https://github.com/OCA/server-ux/tree/11.0/base_tier_validation_formula permite desde la interfaz crear condiciones que se disparan al cambiar de estados e incluso avisar a alguien para que valide y permita pasar al siguiente estado o no. Así se evita tener condiciones tan custom en código y permite mayor flexibilidad.

Reply:
Creo que vi este módulo en las jornadas odoo de este año. Puede ser bastante interesante, por que es verdad que al final tenemos un tocho de código con tropecientas comprobaciones en el flujo de pedido.

Reply:
Sí, es el de las jornadas. NOsotros desde aquellas ya lo usamos en un par de proyectos y funciona bien.